packages/wallets/avana/src/adapter.ts

```diff
@@ -126,7 +128,10 @@ export class AvanaWalletAdapter extends BaseMessageSignerWalletAdapter {
                 throw new WalletPublicKeyError(error?.message, error);
             }
 
-            wallet.on('disconnect', this._disconnected);
+            if (!this._onDisconnectBound) {
+                this._onDisconnectBound = this._disconnected.bind(this);
+                wallet.on('disconnect', this._onDisconnectBound);
+            }
```

Comment:
The reason this listener should never get added twice is because we always check if the adapter is `connected` or `connecting` first in the `connect` function. Only by disconnecting (and thereby removing the listener) can it be added again.